zcash_client_sqlite/src/wallet.rs

```diff
@@ -947,10 +948,11 @@ pub(crate) fn get_target_and_anchor_heights(
     })
 }
 
-fn parse_block_metadata(
-    row: (BlockHeight, Vec<u8>, Option<u32>, Vec<u8>),
+fn parse_block_metadata<P: consensus::Parameters>(
+    params: &P,
+    row: (BlockHeight, Vec<u8>, Option<u32>, Vec<u8>, Option<u32>),
 ) -> Result<BlockMetadata, SqliteClientError> {
-    let (block_height, hash_data, sapling_tree_size_opt, sapling_tree) = row;
+    let (block_height, hash_data, sapling_tree_size_opt, sapling_tree, orchard_tree_size_opt) = row;
     let sapling_tree_size = sapling_tree_size_opt.map_or_else(|| {
         if sapling_tree == BLOCK_SAPLING_FRONTIER_ABSENT {
             Err(SqliteClientError::CorruptedData("One of either the Sapling tree size or the legacy Sapling commitment tree must be present.".to_owned()))
```

Comment:
I do not remember why the `sapling_commitment_tree_size` column in the `blocks` table was made nullable, given that we ran a migration that parsed the legacy commitment tree data into that column (the same data we use here to make `sapling_tree_size` non-optional). Do we still insert the legacy Sapling commitment tree into the database even after shardtree deployment? @nuttycom if you remember why you implemented it this way, a code comment would be very useful.

Reply:
It was left nullable so that wallets connecting to older lightwalletd servers would be able to continue to function, although they would have been restricted to scanning in linear order. However, since the decision to implement it that way, we decided that connecting to an older lightwalletd is an unsupported use case; however, I'm not sure that it's worth the effort of writing a migration, particularly since Orchard tree sizes being null _is_ a supported use case, and I prefer to have the symmetry. That potentially means that we could get rid of the `CorruptedData` error here, and instead just return `None` in this case, since we have to handle the absence of the tree size at the use site (so that the error path is the same as for Orchard) anyway.